Assets/Scripts/MazeGenerator.cs

```diff
@@ -0,0 +1,177 @@
+﻿using System.Collections;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class MazeGenerator : MonoBehaviour
+{
+    [SerializeField]
+    private int width;
+
+    [SerializeField]
+    private int height;
+
+    public Transform Brick;
+    private int[,] Maze;
+    private List<Vector3> pathMazes = new List<Vector3>();
+    private Stack<Vector2> _tiletoTry = new Stack<Vector2>();
+    private List<Vector2> offsets = new List<Vector2> { new Vector2(0, 1), new Vector2(0, -1), new Vector2(1, 0), new Vector2(-1, 0) };
```

Comment:
this line is too long and should be multi lined. Also, you should make this a readonly or const since it looks like it won't be modified